### PR TITLE
Support restoring to a specific time

### LINKED ISSUE #163: Allow point in time restoration using a specific datetime
### Describe the Feature

AWS Console and AWS's vanilla `aws_rds_cluster` resource allows for specifying a date time as opposed to using the latest restorable time.

### Expected Behavior

Have the option to pass in `restore_to_time` as a UTC datetime string instead of passing `use_latest_restorable_time` (or passing it as `false`)

### Use Case

Having this option is really valuable for running Data Recovery following an incident where the latest restorable time's data may be corrupt.

### Describe Ideal Solution

Have a new RDS Cluster created using restored data from a particular point in time (not necessarily the latest point in time).

### Alternatives Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ericluria**
## what
- Adds `restore_to_time` as an attribute of the `restore_to_point_in_time` parameter, allowing to restore to a specific datetime rather than the latest restorable time.

## why
- Fixes missing functionality that's [supported by the `aws_rds_cluster` resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster#restore_to_time).

## references
- Closes #163 

## questions
~I noticed while making this PR that `source_cluster_identifier` defaults to `"120m"`, which looks like an inadvertent copy-and-paste from another parameter. In the `aws_rds_cluster` resource, this parameter is required, so it doesn't make sense to have a default anyway. Would you recommend leaving this parameter as-is, or should we fix it?~

edit: fixed as part of this PR

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -200,9 +200,25 @@ variable "restore_to_point_in_time" {
...
+    `restore_type`:
+      Type of restore to be performed. Valid options are "full-copy" and "copy-on-write".
+      default: "copy-on-write"
+    `use_latest_restorable_time`:
+      Set to true to restore the database cluster to the latest restorable backup time. Conflicts with `restore_to_time`.
+      default: true
+    `restore_to_time`:
+      Date and time in UTC format to restore the database cluster to. Conflicts with `use_latest_restorable_time`.
+      default: null
+EOT
```
Comment (milldr) [inline variables.tf:221]: Please make these object inputs optional and match the description. For example

```hcl
variable "restore_to_point_in_time" {
  type = list(object({
    source_cluster_identifier  = optional(string, "120m")
    restore_type               = optional(string, "copy-on-write")
    use_latest_restorable_time = optional(bool, true)
    restore_to_time            = optional(string, null)
  }))
```

Comment (milldr) [reply variables.tf:221]: Plus, we can also refactor the dynamic block in `main.tf`

```hcl
  dynamic "restore_to_point_in_time" {
    for_each = var.restore_to_point_in_time
    content {
      source_cluster_identifier  = restore_to_point_in_time.valuesource_cluster_identifier
      restore_type               = restore_to_point_in_time.valuerestore_type
      use_latest_restorable_time = restore_to_point_in_time.value.use_latest_restorable_time
      restore_to_time            = restore_to_point_in_time.value.restore_to_time
    }
  }
  ```

Comment (ericluria) [reply variables.tf:221]: Thanks! Was going to ask about that - pushing up the change in a sec

**Review by milldr — APPROVED** [review]
LGTM 🚀

**Pull request merged**